tools/outline_proxy_controller/outline_controller_server.cpp

```diff
@@ -144,8 +149,13 @@ OutlineControllerServer::OutlineControllerServer(boost::asio::io_context& io_con
   ::unlink(unix_socket_name.c_str());
   boost::asio::spawn(io_context, [&](boost::asio::yield_context yield) {
     stream_protocol::acceptor acceptor(io_context, stream_protocol::endpoint(unix_socket_name));
-    chmod(unix_socket_name.c_str(),
-          S_IRWXU | S_IROTH | S_IWOTH);  // enables all user to read from and write into socket
+    auto outlineGrp = getgrnam(OUTLINE_GRP_NAME);
+    if (outlineGrp != nullptr) {
+      chown(unix_socket_name.c_str(), -1, outlineGrp->gr_gid);
```

Comment:
As discussed on chat, let's make the user the owner, so they don't need to restart.